network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/revoked/AccessRevokedDialog.kt

```diff
@@ -17,37 +17,77 @@
 package com.duckduckgo.networkprotection.impl.revoked
 
 import android.app.Activity
+import android.content.SharedPreferences
+import androidx.core.content.edit
 import com.duckduckgo.app.di.AppCoroutineScope
 import com.duckduckgo.common.ui.view.dialog.TextAlertDialogBuilder
 import com.duckduckgo.common.ui.view.dialog.TextAlertDialogBuilder.EventListener
 import com.duckduckgo.common.utils.DispatcherProvider
 import com.duckduckgo.di.scopes.AppScope
+import com.duckduckgo.mobile.android.vpn.prefs.VpnSharedPreferencesProvider
 import com.duckduckgo.navigation.api.GlobalActivityStarter
 import com.duckduckgo.networkprotection.impl.R
 import com.duckduckgo.networkprotection.impl.pixels.NetworkProtectionPixels
-import com.duckduckgo.networkprotection.impl.store.NetworkProtectionRepository
 import com.duckduckgo.subscriptions.api.SubscriptionScreens.SubscriptionScreenNoParams
 import com.squareup.anvil.annotations.ContributesBinding
 import javax.inject.Inject
 import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
 
 interface AccessRevokedDialog {
-    fun show(activity: Activity)
+    /**
+     * Call this method to always show the dialog
+     */
+    fun showAlways(activity: Activity)
```

Comment:
We currently don't use `showAlways` but left it for completion, I could also remove it though